sds_data_manager/stacks/processing_stack.py

```diff
@@ -77,47 +80,53 @@ def __init__(
             db_secret_name=db_secret_name,
         )
 
-        self.instrument_lambda = InstrumentLambda(
-            self,
-            "InstrumentLambda",
-            processing_step_name=processing_step_name,
-            data_bucket=data_bucket,
-            code_path=str(lambda_code_directory),
-            instrument_target=instrument_target,
-            instrument_sources=instrument_sources,
-            db_secret_name=db_secret_name,
-            rds_security_group=rds_security_group,
-            subnets=subnets,
-            vpc=vpc,
-        )
-
         self.step_function = SdcStepFunction(
             self,
             f"SdcStepFunction-{processing_step_name}",
             processing_step_name=processing_step_name,
-            processing_system=self.instrument_lambda,
             batch_resources=self.batch_resources,
-            instrument_target=instrument_target,
             data_bucket=data_bucket,
             db_secret_name=db_secret_name,
+            dependents=dependents,
         )
 
-        # TODO: This will be a construct and also we will add to its capabilities.
-        rule = events.Rule(
+        self.instrument_lambda = InstrumentLambda(
             self,
-            "rule",
-            event_pattern=events.EventPattern(
-                source=["aws.s3"],
-                detail_type=["Object Created"],
-                detail={
-                    "bucket": {"name": [data_bucket.bucket_name]},
-                    "object": {"key": [{"prefix": f"{instrument_sources}"}]},
-                },
-            ),
+            "InstrumentLambda",
+            data_bucket=data_bucket,
+            code_path=str(lambda_code_directory),
+            instrument=instrument,
+            instrument_dependents=instrument_dependents,
+            step_function_arn=self.step_function.state_machine.state_machine_arn,
+            step_function_policy=self.step_function.execution_policy,
+            db_secret_name=db_secret_name,
+            rds_security_group=rds_security_group,
+            subnets=subnets,
+            db_secret_arn=db_secret_arn,
+            vpc=vpc,
         )
 
-        rule.add_target(event_targets.SfnStateMachine(self.step_function.state_machine))
+        # Kicks off Step Function as a result of object ingested into directories in
+        # s3 bucket (instrument_sources).
+        # TODO: Right now these directories are created manually.
+        #  Add code so that they are not.
+        for source in instrument_dependents:
+            rule = events.Rule(
+                self,
+                f"Rule-{processing_step_name}-{source}",
+                event_pattern=events.EventPattern(
+                    source=["aws.s3"],
+                    detail_type=["Object Created"],
+                    detail={
+                        "bucket": {"name": [data_bucket.bucket_name]},
+                        "object": {"key": [{"prefix": f"{instrument}_{source}"}]},
```

Comment:
```suggestion
                        "object": {"key": [{"prefix": f"{instrument}_{source}"} for source in instrument_dependents]},
```
I think you could squash all of the rules down into a single one with multiple prefixes like this? Putting the for-loop into the list of prefixes instead of outside.